onnxruntime/core/graph/conv_activation_fusion.cc

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/initializer.h"
+#include "core/graph/conv_activation_fusion.h"
+
+using namespace onnx;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+
+namespace {
+bool IsFusableActivation(const Node& node) {
+  const std::string& op_type = node.OpType();
+  return op_type == "LeakyRelu" || op_type == "Relu" || op_type == "Sigmoid" || op_type == "Tanh";
+}
+}  // namespace
+
+Status ConvActivationFusion::Apply(Graph& graph, bool& modified) const {
+  GraphViewer graph_viewer(graph);
+  const auto& order = graph_viewer.GetNodesInTopologicalOrder();
+
+  std::vector<onnxruntime::NodeIndex> removed_nodes;
+  for (auto index : order) {
+    auto node = graph.GetNode(index);
+    if (node->OpType() != "Conv" || node->GetOutputEdgesCount() != 1) {
+      continue;
+    }
+    const Node& next_node = *(node->OutputNodesBegin());
+    if (!IsFusableActivation(next_node) || graph.IsNodeOutputsInGraphOutputs(next_node)) {
+      continue;
+    }
+
+    Node* conv_node = node;
+    const Node& act_node = next_node;
+    std::vector<NodeArg> input_args, output_args;
+
+    Node& fused_conv = graph.AddNode("fused " + conv_node->Name(), "FusedConv",
+                                     "fused Conv " + conv_node->Name() + "with activation " + act_node.OpType(),
+                                     conv_node->MutableInputDefs(),
+                                     conv_node->MutableOutputDefs(),
+                                     &conv_node->GetAttributes(),
+                                     "com.microsoft");
+
+    //Add a new attribute to specify the activation type
+    AttributeProto act;
+    act.set_name("activation");
+    act.set_type(ONNX_NAMESPACE::AttributeProto_AttributeType_STRING);
+    act.set_s(act_node.OpType());
```

Comment:
DML uses Node->AddAttribute("activation", "string") which internally does all this stuff.

Reply:
Replaced.